spec/controllers/posts_controller_spec.rb

```diff
@@ -0,0 +1,39 @@
+require 'rails_helper'
+
+RSpec.describe PostsController, type: :request do
+  describe 'GET #index' do
+    let :user do
+      User.create do |user|
+        user.name = 'John Doe'
+      end
+    end
+
+    it 'returns http success' do
+      get user_posts_path(user)
+      expect(response).to be_successful
+      expect(response.body).to include('List of user\'s posts')
+    end
+  end
+
+  describe 'GET #show' do
+    let :user do
+      User.create do |user|
+        user.name = 'John Doe'
+      end
+    end
+
+    let :post do
+      Post.create do |post|
+        post.title = 'Post 1'
+        post.text = 'Lorem ipsum dolor sit amet, consectetur adipiscing elit.'
+        post.author = user
+      end
+    end
+
+    it 'returns http success' do
+      get user_post_path(user, post)
+      expect(response).to be_successful
+      expect(response.body).to include('Post\'s details')
+    end
+  end
+end
```

Comment:
- [ ] Kindly make sure to test if the correct template was rendered in both the `show` and `index` actions, as per [this](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/controller_specs_project.md#:~:text=If%20a%20correct%20template%20was%20rendered.) requirement.